xarray/core/combine.py

```diff
@@ -558,8 +558,67 @@ def vars_as_keys(ds):
     return tuple(sorted(ds))
 
 
-def combine_by_coords(
+def _combine_single_variable_hypercube(
     datasets,
+    fill_value=dtypes.NA,
+    data_vars="all",
+    coords="different",
+    compat="no_conflicts",
+    join="outer",
+    combine_attrs="no_conflicts",
+):
+    """
+    Attempt to combine a list of Datasets into a hypercube using their
+    coordinates.
+
+    All provided Datasets must belong to a single variable, ie. must be
+    assigned the same variable name. This precondition is not checked by this
+    function, so the caller is assumed to know what it's doing.
+
+    This function is NOT part of the public API.
+    """
+    if len(datasets) == 0:
+        raise ValueError(
+            "At least one Dataset is required to resolve variable names "
+            "for combined hypercube."
+        )
```

Comment:
Can this Error ever be reached? Wouldn't the 

```python
if not data_objects:
        return Dataset()
```
stop this from being triggered?

Reply:
I intended this check to be part of the internal contract for `_combine_single_variable_hypercube` since it doesn't make sense to build an empty hypercube with this method. There would be no way to determine the variable involved, so a default sentinel name would have to be introduced.